bigchaindb/core.py

```diff
@@ -274,8 +277,27 @@ def get_status(self, txid):
             or 'backlog'). If no transaction with that `txid` was found it
             returns `None`
         """
-        _, status = self.get_transaction(txid, include_status=True)
-        return status
+        try:
+            statuses = self.get_blocks_status_containing_tx(txid).values()
+        except (TypeError, AttributeError):
+            logger.debug(
+                'No blocks found for transaction %s. Looking into BACKLOG.',
+                txid,
+            )
+            if backend.query.\
+                    get_transaction_from_backlog(self.connection, txid):
+                return self.TX_IN_BACKLOG
+
+        else:
+            if self.BLOCK_VALID in statuses:
+                return self.BLOCK_VALID
+            elif self.BLOCK_UNDECIDED in statuses:
+                return self.BLOCK_UNDECIDED
+            elif backend.query.get_transaction_from_backlog(
+                                                        self.connection, txid):
+                return self.TX_IN_BACKLOG
+            elif self.BLOCK_INVALID in statuses:
+                return self.BLOCK_INVALID
```

Comment:
Wondering if this "upgrading" behaviour should be mentioned in the HTTP API docs; that a transaction will always be reported as, e.g., "valid" when there also exists a transaction in an "undecided", "backlog", or "invalid" block.